### PR TITLE
feat: add wrapper Engine

### PR DESCRIPTION
**Pull request by jonathanlee46**
Initial scaffolding for wrapper Engine that will take in JSON and options to be used as input for enquirer.

### CODE REVIEW

**Comment by jonathanlee46** [comment]
Need to figure out what to do with other metadata and when/where to send it, like `surveyId`.

File: src/engine/Engine.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { prompt } from 'enquirer'
+import convertToOutput from './OutputAdapter'
+
+interface settingsJsonTypes {
+  id: string,
+  type: string,
+  properties: object
```
Comment (sumwatshade) [inline src/engine/Engine.ts]: Let's export these types from `src/types/Engine.ts` and include the nested properties more explicitly:

```ts
interface ISurveyPromptSettings { /* ... */ }

interface ISurveySchema {
  id: string,
  createdAt: Date,
  prompts: ISurveyPromptSettings[]
}
```

File: src/engine/Engine.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+class Engine {
+  constructor(options: object) {
```
Comment (sumwatshade) [inline src/engine/Engine.ts]: This should have a type, `IEngineOptions`. You can also store it as `this.options` using Typescript shorthand:

```ts
	constructor(private readonly options: IEngineOptions) {}
```

File: src/engine/Engine.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { prompt } from 'enquirer'
+import convertToOutput from './OutputAdapter'
+
+interface settingsJsonTypes {
+  id: string,
+  type: string,
+  properties: object
+}
+
+class Engine {
```
Comment (sumwatshade) [inline src/engine/Engine.ts]:
```suggestion
export class EnquirerEngine {
```

File: src/engine/Engine.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    const engineOutput = await prompt(surveySettings)
+    const format = this.options?.outputFormat
+    return convertToOutput({
+      engineOutput,
+      format
+    })
+  }
+}
+
+export default Engine
```
Comment (sumwatshade) [inline src/engine/Engine.ts]:
```suggestion
export default EnquirerEngine
```

Comment (sumwatshade) [inline src/engine/Engine.ts]:
```suggestion
export default EnquirerEngine
```

**Review by sumwatshade — CHANGES_REQUESTED** [review]
Can we add at least a unit test that validates a prompt is shown? Or at least that `prompt` is executed with expected options.

**Comment by sumwatshade** [comment]
:rocket: PR was released in [`v0.2.0`](https://github.com/sumwatshade/enquirer-engine/releases/tag/v0.2.0) :rocket:

**Pull request merged**